## Group Discussion
### Using transcriptomics to study developmental in non-model systems

Transcriptomics has opened up the possibility of studying development and developmental plasticity in non-model organisms. For this discussion, we will read some recent papers and discuss how -omics have been used to answer biological questions.

---
## Papers

* Casasa, S., Zattara, E.E. & Moczek, A.P. (2020). Nutrition-responsive gene expression and the developmental evolution of insect polyphenism. Nature Ecology & Evolution, 4, 1–12. [PDF](./Casasa.pdf){:target="_blank"}

* Liedtke, H.C., Harney, E. & Gomez-Mestre, I. (2021). Cross-species transcriptomics uncovers genes underlying genetic accommodation of developmental plasticity in spadefoot toads. Molecular Ecology, 30, 2220–2234. [PDF](./Liedtke.pdf){:target="_blank"}

* Wollenberg Valero, K.C., Garcia-Porta, J., guez, A.R.I., Arias, M.O.N., Shah, A., Randrianiaina, R.-D., Brown, J.L., Glaw, F., Amat, F., nzel, S.K.U., Metzler, D., Isokpehi, R.D. & Vences, M. (2017). Transcriptomic and macroevolutionary evidence for phenotypic uncoupling between frog life history phases. Nature Communications, 8, 1–9. [PDF](./Wollenberg.pdf){:target="_blank"}

* Zhu, W., Liu, L., Wang, X., Gao, X., Jiang, J. & Bin Wang. (2019). Transcriptomics reveals the molecular processes of light-induced rapid darkening of the non-obligate cave dweller Oreolalax rhodostigmatus (Megophryidae, Anura) and their genetic basis of pigmentation strategy. BMC Genomics, 19, 1–13. [PDF](./Zhu.pdf){:target="_blank"}

---

## Focus on:

*	What is the biological question/aim of the paper?
* How is transcriptomics (or other -omics) useful for answering this aims?
* What is the experimental design:
  * What tissues?
  * what treatments?
  * how many biological replicates?
  * Is there a "control" vs. "treatment", or "phenotype 1" vs "phenotype 2"?
* Is there a reference genome available?
* What are some methodological aspects for the nucleic acid extraction and sequencing methods?
* What software/methods are used? For:
  * Transcriptome assembly
  * Reads quantification
  * Differential gene expression
  * Functional enrichment
* How are genes of interest annotated?
